contracts/toCKB-typescript/src/switch/mint_xt/mod.rs

```diff
@@ -1,7 +1,299 @@
+    if input_xt_num != 0 {
+        return Err(Error::InvalidXTInInputOrOutput);
+    }
+    let output_xt_num = QueryIter::new(load_cell_type, Source::Output)
+        .filter(|type_opt| type_opt.is_some())
+        .map(|type_opt| type_opt.unwrap())
+        .filter(|script| {
+            script.code_hash().raw_data().as_ref() == SUDT_CODE_HASH.as_ref()
+                && script.args().raw_data().as_ref() == lock_hash.as_ref()
+        })
+        .count();
+    debug!("output_xt_num: {}", output_xt_num);
+    if output_xt_num != 2 {
+        return Err(Error::InvalidXTInInputOrOutput);
+    }
+    let xt_amount = data.get_btc_lot_size()?.get_sudt_amount();
+    debug!("xt_amount: {}", xt_amount);
+    // fixed order of output cells is required
+    // user-sudt-cell should be outputs[1]
+    // signer-sudt-cell should be outputs[2]
+    let expect = [
+        (
+            1,
+            data.user_lockscript.as_ref(),
+            xt_amount - xt_amount * SIGNER_FEE_RATE.0 / SIGNER_FEE_RATE.1,
+        ),
+        (
+            2,
+            data.signer_lockscript.as_ref(),
+            xt_amount * SIGNER_FEE_RATE.0 / SIGNER_FEE_RATE.1,
+        ),
+    ];
+    debug!("expect: {:?}", expect);
+
+    for (i, lockscript, amout) in expect.iter() {
+        let script = load_cell_type(*i, Source::Output)?;
+        if script.is_none() {
+            return Err(Error::InvalidMintOutput);
+        }
+        let script = script.unwrap();
+        if !(script.code_hash().raw_data().as_ref() == SUDT_CODE_HASH.as_ref()
+            && script.args().raw_data().as_ref() == lock_hash.as_ref()
+            && script.hash_type() == 0u8.into())
+        {
+            return Err(Error::InvalidMintOutput);
+        }
+        let cell_data = load_cell_data(*i, Source::Output)?;
+        let mut amount_vec = [0u8; 16];
+        amount_vec.copy_from_slice(&cell_data);
+        let token_amount = u128::from_le_bytes(amount_vec);
+        debug!("token_amount: {}, amout: {}", token_amount, amout);
+        if token_amount != *amout {
+            return Err(Error::InvalidMintOutput);
+        }
+        let lock = load_cell_lock(*i, Source::Output)?;
+        debug!(
+            "lock: {:?}, expect lock: {:?}",
+            hex::encode(lock.as_slice()),
+            hex::encode(lockscript.as_ref())
+        );
+        if lock.as_slice() != lockscript.as_ref() {
+            return Err(Error::InvalidMintOutput);
+        }
+    }
+    Ok(())
+}
+
+pub fn verify(toCKB_data_tuple: &ToCKBCellDataTuple) -> Result<(), Error> {
+    debug!("start mint_xt");
+    let input_data = toCKB_data_tuple.0.as_ref().expect("should not happen");
+    let output_data = toCKB_data_tuple.1.as_ref().expect("should not happen");
+    verify_data(input_data, output_data)?;
+    debug!("verify data finish");
+    verify_witness(input_data)?;
+    debug!("verify witness finish");
+    verify_xt_issue(input_data)?;
+    debug!("verify xt issue finish");
     Ok(())
 }
```

Comment:
Missing capacity check of toCKB_cell on tx.outputs  ?

Reply:
> should check sth. to ensure the tx has no other outputs. Someone may generate udt outputs or other sudt outputs which code_hash != SUDT_CODE_HASH in the config.rs. It may do harm

This check is in `verify_xt_issue` function.

Reply:
> Missing capacity check of toCKB_cell on tx.outputs ?

@mkxbl Should this be added to switch and make every mod reuse it?

Reply:
don't think so. 

some txs have different capacity verify logic, and some verification is contextual